src/test/txvalidationcache_tests.cpp

```diff
@@ -109,10 +109,15 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
 static void ValidateCheckInputsForAllFlags(const CTransaction &tx, uint32_t failing_flags, bool add_to_cache) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
     PrecomputedTransactionData txdata;
-    // If we add many more flags, this loop can get too expensive, but we can
-    // rewrite in the future to randomly pick a set of flags to evaluate.
-    for (uint32_t test_flags=0; test_flags < (1U << 16); test_flags += 1) {
+
+    FastRandomContext insecure_rand(true);
+
+    for (int count=0; count < 10000; count++) {
         TxValidationState state;
+
+        // Randomly selects flag combinations
+        uint32_t test_flags = (uint32_t) insecure_rand.randrange(SCRIPT_VERIFY_HIGHEST_FLAG << 1);
```

Comment:
Using the marker style, to get the same you'd `randrange((SCRIPT_VERIFY_END_MARKER - 1) << 1)` here.

However, I'm not sure why you are doing `<< 1` here (`1U << 17`), as the original code only goes up to `SCRIPT_VERIFY_END_MARKER - 1` (i.e. `(1U << 16)`). To retain the same, this would be `randrange(SCRIPT_VERIFY_END_MARKER - 1);`

Reply:
Hi, @kalewoof. Thanks for reviewing.
The reason for that is that we want to possibly test all flag combinations (or a random subset thereof). The lowest value possible with all flags off is 0, the highest would be the sum of all constants. So if the highest constant is 1<<16, having the maximum value used for test be 1<<16 would preclude testing combinations of that flag being used. We need to test from 0 to `1111111111111111` (1<<17 -1). Because of the range being exclusive of the highest value, I pass `1<<17 = (SCRIPT_VERIFY_END_MARKER - 1) << 1`
Hope that clarifies.

Reply:
The test currently only going to 1<<16 is probably a mistake caused by adding a flag but not updating the test.

Reply:
See https://github.com/bitcoin/bitcoin/commit/9dabfe49c066301ef75bcfcb089fd308366127c4

Reply:
Test created in https://github.com/bitcoin/bitcoin/commit/e3f9c05b966622146e090f2a01a913516ccb874a. Not updated after the new flag added in https://github.com/bitcoin/bitcoin/commit/9dabfe49c066301ef75bcfcb089fd308366127c4